File: examples/tentacles/hello-world/cpp/handler.cpp
#include <iostream>
#include <string>
int main() {
    std::string s((std::istreambuf_iterator<char>(std::cin)), std::istreambuf_iterator<char>());
    std::string name = "world";
    auto k = s.find("\"name\"");
    if (k != std::string::npos) {
        auto q = s.find('"', k + 6);
        if (q != std::string::npos) {
            auto q2 = s.find('"', q + 1);
            if (q2 != std::string::npos) name = s.substr(q + 1, q2 - q - 1);
        }
    }
    std::cout << "{\"v\":1,\"ok\":true,\"result\":{\"arm\":\"hello-cpp\",\"lang\":\"cpp\",\"hello\":\"hello, " << name << "\"}}";
    return 0;
}

File: examples/tentacles/hello-world/rust/handler.rs
use std::io::{self, Read};
fn main() {
    let mut s = String::new();
    let _ = io::stdin().read_to_string(&mut s);
    let name = extract(&s).unwrap_or("world");
    print!("{{\"v\":1,\"ok\":true,\"result\":{{\"arm\":\"hello-rs\",\"lang\":\"rust\",\"hello\":\"hello, {}\"}}}}", name);
}
fn extract(s: &str) -> Option<&str> {
    let key = "\"name\"";
    let i = s.find(key)?;
    let rest = &s[i + key.len()..];
    let q = rest.find('"')?;
    let rest = &rest[q + 1..];
    let q2 = rest.find('"')?;
    Some(&rest[..q2])
}
